File: frontend/src/types/lead.ts
export enum LeadStatus {
  New = 0,
  Contacted = 1,
  Qualified = 2,
  Unqualified = 3,
  Converted = 4
}

export enum LeadSource {
  Website = 0,
  Referral = 1,
  Campaign = 2,
  SocialMedia = 3,
  Event = 4,
  Other = 5
}

export interface Lead {
  id: string
  companyId?: string
  contactId?: string
  title: string
  description?: string
  source: string
  status: string
  value?: number
  probability?: number
  expectedCloseDate?: string
  assignedUserId?: string
  notes?: string
  isActive: boolean
  createdAt: string
  updatedAt?: string
  // Navigation properties (optional, may be populated by backend)
  company?: any
  contact?: any
  assignedUser?: any
}

export interface CreateLeadDto {
  companyId?: string
  contactId?: string
  title: string
  description?: string
  source: string
  status?: string
  value?: number
  probability?: number
  expectedCloseDate?: string
  assignedUserId?: string
  notes?: string
  isActive?: boolean
}

export interface UpdateLeadDto {
  companyId?: string
  contactId?: string
  title?: string
  description?: string
  source?: string
  status?: string
  value?: number
  probability?: number
  expectedCloseDate?: string
  assignedUserId?: string
  notes?: string
  isActive?: boolean
}

export interface LeadsResponse {
  items: Lead[]
  totalCount: number
  pageNumber: number
  pageSize: number
  totalPages: number
}

File: backend/Core/CRM.Domain/Entities/Lead.cs
namespace CRM.Domain.Entities;

public class Lead : BaseEntity
{
    public Guid? CompanyId { get; set; }
    public Guid? ContactId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Source { get; set; }
    public string Status { get; set; } = "new";
    public decimal? Value { get; set; }
    public int? Probability { get; set; }
    public DateTime? ExpectedCloseDate { get; set; }
    public Guid? AssignedUserId { get; set; }
    public string? Notes { get; set; }
    
    // Navigation properties
    public Company? Company { get; set; }
    public Contact? Contact { get; set; }
    public User? AssignedUser { get; set; }
    public ICollection<Activity> Activities { get; set; } = new List<Activity>();
}
